File: hw3/WebTest/src/main/java/ru/hse/lupuleac/webtest/elements/PageElement.java
package ru.hse.lupuleac.webtest.elements;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
public abstract class PageElement {
    protected WebDriver webDriver;
    protected WebElement webElement;

    public PageElement(WebDriver driver, By by) {
        webDriver = driver;
        webElement = driver.findElement(by);
    }
}

File: hw3/WebDriverTest/WebDriverTest/Elements/PageElement.cs
using OpenQA.Selenium;

namespace WebDriverTest.Elements
{
    public abstract class PageElement
    {
        protected IWebDriver WebDriver;
        protected IWebElement WebElement;

        protected PageElement(IWebDriver webDriver, By by)
        {
            WebDriver = webDriver;
            WebElement = webDriver.FindElement(by);
        }
    }

    public class ButtonElement : PageElement
    {
        public ButtonElement(IWebDriver webDriver, By by) : base(webDriver, by)
        {
        }

        public void Click()
        {
            WebElement.Click();
        }
    }

    public class TextElement : PageElement
    {
        public TextElement(IWebDriver webDriver, By by) : base(webDriver, by)
        {
        }

        public string Text
        {
            get => WebElement.Text;
            set => WebElement.SendKeys(value);
        }
    }
}